nevergrad/functions/functionlib.py

```diff
@@ -210,6 +216,8 @@ def __init__(  # pylint: disable=too-many-arguments
             useful_dimensions=block_dimension * num_blocks,
             discrete=any(x in name for x in ["onemax", "leadingones", "jump"]),
         )
+        if self._bounded:
+            self._trs = trs.ArctanBound(0, 1)
```

Comment:
Let's make the name (a little bit) more explicit, and attributes need to be defined in all cases
```suggestion
        self._transf: tp.Optional[trs.BoundTransform] = trs.ArctanBound(0, 1) if bounded else None
```